[DEV] Build failing? org.openhab.binding.amazondashbutton

Hi everyone,
do I have a local probleme here, or is the build failing?

I'm on openhab/openhab2-addons commit id 17aa3a218cfbfe670bd0b30f2b69995d9da0905b, and latest eclipse/smarthome master and openhab/openhab-core (which builds fine).

Maven error (mvn clean install):

```
[INFO] ------------------------------------------------------------------------
[INFO] Building Amazon Dash Button Binding 2.2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ org.openhab.binding.amazondashbutton ---
[INFO] Deleting /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/target
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:build-qualifier (default-build-qualifier) @ org.openhab.binding.amazondashbutton ---
[INFO] The project's OSGi version is 2.2.0.201709201846
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-id (default-validate-id) @ org.openhab.binding.amazondashbutton ---
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-version (default-validate-version) @ org.openhab.binding.amazondashbutton ---
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:resources (default-resources) @ org.openhab.binding.amazondashbutton ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/resources
[INFO] 
[INFO] --- tycho-compiler-plugin:1.0.0:compile (default-compile) @ org.openhab.binding.amazondashbutton ---
[INFO] Compiling 12 source files to /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] openHAB 2 Add-ons .................................. SUCCESS [  0.913 s]
[INFO] openHAB Add-ons .................................... SUCCESS [  0.037 s]
[INFO] openHAB Bindings ................................... SUCCESS [  0.035 s]
[INFO] AirQuality Binding ................................. SUCCESS [ 32.029 s]
[INFO] AllPlay Binding .................................... SUCCESS [ 15.123 s]
[INFO] Amazon Dash Button Binding ......................... FAILURE [  0.584 s]
[... more skipped ...]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:39 min
[INFO] Finished at: 2017-09-20T20:48:26+02:00
[INFO] Final Memory: 107M/863M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.0.0:compile (default-compile) on project org.openhab.binding.amazondashbutton: Compilation failure: Compilation failure: 
[ERROR] /home/holger/DATA/dev/eclipse4.7_openhab/openhab2-master/git/openhab2-addons/addons/binding/org.openhab.binding.amazondashbutton/src/main/java/org/openhab/binding/amazondashbutton/internal/AmazonDashButtonHandlerFactory.java:[46] 
[ERROR] 	return null;
[ERROR] 	       ^^^^
[ERROR] Null type mismatch: required '@NonNull ThingHandler' but the provided value is null
[ERROR] 1 problem (1 error)
[ERROR] -> [Help 1]
```

If this is a local problem, how can I fix it? Am a bit lost right now...

Thanks!

--- a/addons/binding/org.openhab.binding.amazondashbutton/.settings/org.eclipse.jdt.core.prefs
+++ b/addons/binding/org.openhab.binding.amazondashbutton/.settings/org.eclipse.jdt.core.prefs
@@ -1,7 +1,7 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.7
-org.eclipse.jdt.core.compiler.compliance=1.7
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.compliance=1.8
 org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
 org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
-org.eclipse.jdt.core.compiler.source=1.7
+org.eclipse.jdt.core.compiler.source=1.8

--- a/addons/binding/org.openhab.binding.max.test/.settings/org.eclipse.jdt.core.prefs
+++ b/addons/binding/org.openhab.binding.max.test/.settings/org.eclipse.jdt.core.prefs
@@ -1,8 +1,8 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.7
-org.eclipse.jdt.core.compiler.compliance=1.7
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.compliance=1.8
 org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
 org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
 org.eclipse.jdt.core.compiler.problem.forbiddenReference=warning
-org.eclipse.jdt.core.compiler.source=1.7
+org.eclipse.jdt.core.compiler.source=1.8

--- a/addons/binding/org.openhab.binding.tellstick/.settings/org.eclipse.jdt.core.prefs
+++ b/addons/binding/org.openhab.binding.tellstick/.settings/org.eclipse.jdt.core.prefs
@@ -1,12 +1,12 @@
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
 org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.7
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
 org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
-org.eclipse.jdt.core.compiler.compliance=1.7
+org.eclipse.jdt.core.compiler.compliance=1.8
 org.eclipse.jdt.core.compiler.debug.lineNumber=generate
 org.eclipse.jdt.core.compiler.debug.localVariable=generate
 org.eclipse.jdt.core.compiler.debug.sourceFile=generate
 org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
 org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
-org.eclipse.jdt.core.compiler.source=1.7
+org.eclipse.jdt.core.compiler.source=1.8

